src/generic_test.rs

```diff
@@ -34,14 +36,17 @@ impl Peer<Id> for TestPeer<Id> {
     fn new(id: Id, addr: String) -> TestPeer<Id> {
         TestPeer { id, net_addr: addr }
     }
+
     fn get_id(&self) -> Id {
         self.id.clone()
     }
+
     fn get_net_addr(&self) -> String {
         self.net_addr.clone()
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, Hash, PartialEq, PartialOrd, Ord)]
```

Comment:
The same, why not used traits are added?

Reply:
Same as above.